tom/bot.py

```diff
@@ -2,34 +2,38 @@
 import random
 import json
 import logging as log
+from copy import copy
 
 from tom.github import GitHub, GitHubInterface, PR
 from tom.jenkins import Jenkins
 from tom.slack import CommandDispatcher
-from tom.utils import confirmation
-
-
-class Tom():
-    def __init__(self, secrets, interactive):
-        github = secrets["GITHUB_TOKEN"]
-        self.github = GitHub(github)
-
-        user = secrets["JENKINS_USER"]
-        token = secrets["JENKINS_TOKEN"]
-        if not (user and token):
-            sys.exit("Cannot start Tom without Jenkins credentials")
-        crumb = secrets["JENKINS_CRUMB"]
-        self.jenkins = Jenkins(user, token, crumb)
+from tom.utils import confirmation, pretty
+
+
+class Bot():
+    def __init__(
+            self, username, secrets, orgs, repo_maintainers, jenkins, job, maintainers, trusted,
+            directory, interactive):
+        self.username = username
+        self.secrets = secrets
+        self.orgs = orgs
+        self.repo_maintainers = repo_maintainers
+        self.jenkins = Jenkins(jenkins, job, secrets)
+        self.default_maintainers = maintainers
+        self.trusted = trusted
+        self.directory = directory
+        self.interactive = interactive
+        self.github = GitHub(secrets["GITHUB_TOKEN"])
 
-        self.slack_read_token = secrets["SLACK_READ_TOKEN"]
-        bot_token = secrets["SLACK_SEND_TOKEN"]
-        app_token = secrets["SLACK_APP_TOKEN"]
         self.slack = None
-        if self.slack_read_token and bot_token and app_token:
+        if "SLACK_READ_TOKEN" in secrets:
+            self.slack_read_token = secrets["SLACK_READ_TOKEN"]
+            bot_token = secrets["SLACK_SEND_TOKEN"]
+            app_token = secrets["SLACK_APP_TOKEN"]
```

Comment:
What about checks of the other token's availability? Or maybe just add ``try-except KeyError``?